byzcoin/service_test.go

```diff
@@ -1125,6 +1126,275 @@ func TestService_SetConfigInterval(t *testing.T) {
 	}
 }
 
+func TestService_SetConfigRosterKeepLeader(t *testing.T) {
+	s := newSer(t, 1, testInterval)
+	defer s.local.CloseAll()
+
+	log.Lvl1("Creating blocks to check rotation of the roster while keeping leader")
+	rosterR := s.roster
+	rosterLast := rosterR
+	for i := 0; i < 6; i++ {
+		log.Lvl2("Verifying the correct roster is in place")
+		latest, err := s.service().db().GetLatestByID(s.genesis.Hash)
+		require.Nil(t, err)
+		require.True(t, latest.Roster.ID.Equal(rosterLast.ID), "roster has not been updated")
+		rosterLast = rosterR
+		rosterR = onet.NewRoster([]*network.ServerIdentity{
+			rosterR.List[0], rosterR.List[2], rosterR.List[3], rosterR.List[1]})
+		log.Lvl2("Creating block", i)
+		ctx, _ := createConfigTx(t, testInterval, *rosterR, defaultMaxBlockSize, s)
+		s.sendTxAndWait(t, ctx, 10)
+	}
+	time.Sleep(time.Second)
```

Comment:
bummer.

Reply:
Yes, cleaning up is still messy. So after 2h of following protocol-creation and stuff, sometimes a `time.Sleep` comes in handy ;)

I removed this specific sleep, as it also seems to work without. If Travis complains, I'll add it again!